docs/source/tutorial/type_check.rst

```diff
@@ -0,0 +1,222 @@
+Type check
+----------
+
+.. currentmodule:: chainer
+
+In this section, you will learn about the following things:
+
+* Basic usage of type check
+* Detail of type information
+* Internal mechanism of type check
+* More complicated cases
+* Call functions
+* Typical type check example
+
+After reading this section, you will be able to:
+
+* Write a code to check types of input arguments of your own functions
+
+
+Basic usage of type check
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each implementations of :class:`Function` has a method for type check, :meth:`check_type_forward`.
+This function is called just before :meth:`forward` method of the :class:`Function`.
+You can override this method to check the condition about types of arguments.
+
+:meth:`check_type_forward` gets an argument ``in_types``::
+
+   def check_type_forward(self, in_types):
+     ...
+
+``in_types`` is an instance of :class:`utils.type_check.TypeInfoTuple`, that is a sub-class of ``tuple``.
+To get type information about the first argument, use ``in_types[0]``.
+If the function gets multiple arguments, we recommend to use new variables for readability::
+
+  x_type, y_type = in_types
+
+``x_type`` represents the type of the first argument, and ``y_type`` represents the second one in this case.
+
+We describe usage of ``in_types`` with an example.
+When you want to check if the number of dimention of ``x_type`` equals to ``2``, write this code::
+
+  utils.type_check.expect(x_type.ndim == 2)
+
+When this condition is true, nothing happens.
+Otherwise this code throws an exception, and a user gets a message like this::
+
+  Expect: in_types[0].ndim == 2
+  Actual: 3 != 2
+
+This error message means that "``ndim`` of the first argument expected to be ``2``, but actually it is ``3``".
+
+
+Detail of type information
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can access three information of ``x_type``.
+
+s- ``.shape`` is a ``tuple`` of ``int``. Each value is size of each dimension.
+- ``.ndim`` is ``int`` value representing the number of dimensions. Note that ``ndim == len(shape)``
+- ``.dtype`` is ``numpy.dtype`` representing data type of the value.
+
+You can check all members.
+For exaple, size of first dimension must to be positive, you can write like this::
```

Comment:
For **example**, **the** size of **the** first dimension **must be** positive ...